scripts/quality-gate-staged.py

```
@@ -0,0 +1,382 @@
+                    'cached': False
+                }
+            
+            # If AST-GREP available and no critical issues, try deeper analysis
+            if HAS_AST_GREP and self.registry and not critical_issues:
+                try:
+                    ast_result = self._run_ast_grep_analysis(filepath, content)
+                    # Use AST-GREP score if it found more issues
+                    if ast_result.get('bad_count', 0) > len(bad_patterns):
+                        result = ast_result
+                        result['cached'] = False
+                except:
```

Ruff (0.13.1)

Do not use bare `except`

(E722)